jxscout: init at 0.8.2

fabaff opened the pull request:
Tool to perform JavaScript analysis for security researchers

https://github.com/francisconeves97/jxscout

Related to https://github.com/NixOS/nixpkgs/issues/81418

Closes #403287
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 403921`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>jxscout</li>
  </ul>
</details>

zeshi09 commented:
Friend, tell me, please. When I was trying to build this package, I ran into some problems. I haven't fully figured out how to implement javascript dependencies support when building a package. Since even the installation of dependencies is built into the functionality of the utility. But your version doesn't do that, as I understand it. Is it not necessary to do this? Sorry for the newbie question.
And thank you so much for taking this job.

nicolas-goudry reviewed (changes requested):
## Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] `meta.mainProgram` is set, if applicable.
- [ ] ~~build time only dependencies are declared in `nativeBuildInputs`~~
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [ ] ~~when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.~~
- [ ] ~~patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed~~
- [ ] ~~patches that are remotely available are fetched rather than vendored~~

## Possible improvements

See suggested changes

## Comments

As per what @zeshi09 said, I built the binary successfully and can see the [requested JS files](https://github.com/francisconeves97/jxscout/blob/v0.6.0/Makefile#L11-L12) in upstream repository. Hence, JS should not be needed to build the program.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>jxscout</li>
  </ul>
</details>

nicolas-goudry commented on pkgs/by-name/jx/jxscout/package.nix:
```diff
@@ -0,0 +1,33 @@
+    changelog = "https://github.com/francisconeves97/jxscout/releases/tag/${src.tag}";
+    license = lib.licenses.gpl3Only;
+    maintainers = with lib.maintainers; [ fab ];
+    mainProgram = "jxscout";
```
```suggestion
    mainProgram = "jxscout";
    platforms = lib.platforms.all;
```

Aleksanaa replied:
`buildGoModule` sets available platforms by default

nicolas-goudry replied:
Neat, didn’t know that.

Aleksanaa commented on pkgs/by-name/jx/jxscout/package.nix:
```diff
@@ -0,0 +1,33 @@
+    hash = "sha256-MMTMAdj+N0jgwWnSOoBSGCyGITugtAcK/4SQAMjUcaU=";
+  };
+
+  vendorHash = null;
```
If it's okay to use vendored modules from upstream (and they are complete), we can set `vendorHash = null`.

https://nixos.org/manual/nixpkgs/unstable/#var-go-vendorHash

nicolas-goudry replied:
@Aleksanaa Thanks for making me aware of this!

zeshi09 commented:
New version of this tool is available https://github.com/francisconeves97/jxscout/releases/tag/v0.8.2
And i should to note this build doesn't work with your code
```
error: builder for '/nix/store/a5cj2f1s7gbjc1vv3vjnhinvymn5j3vr-jxscout-0.8.2.drv' failed with exit code 1;
       last 25 log lines:
       >                                    - }
       >                                     +([]astanalyzer.ASTAnalyzerTreeNode) {
       >                                   }
       >              Test:           TestFormatMatchesV1/behavior_with_misc
       > --- FAIL: TestFormatMatchesV1WithExampleData (0.00s)
       >     format_test.go:287:
       >             Error Trace:    /build/source/internal/modules/ast-analyzer/format_test.go:287
       >          Error:          Not equal:
       >                              expected: 30
       >                            actual  : 2
       >             Test:           TestFormatMatchesV1WithExampleData
       >              Messages:       Expected 30 matches in the result
       >     format_test.go:307:
       >               Error Trace:    /build/source/internal/modules/ast-analyzer/format_test.go:307
       >          Error:          Should be true
       >          Test:           TestFormatMatchesV1WithExampleData
       >              Messages:       Expected category Iframe Communication to be present
       >     format_test.go:307:
       >            Error Trace:    /build/source/internal/modules/ast-analyzer/format_test.go:307
       >          Error:          Should be true
       >          Test:           TestFormatMatchesV1WithExampleData
       >              Messages:       Expected category Behavior to be present
       > FAIL
       > FAIL      github.com/francisconeves97/jxscout/internal/modules/ast-analyzer       0.010s
       > FAIL
       For full logs, run:
         nix log /nix/store/a5cj2f1s7gbjc1vv3vjnhinvymn5j3vr-jxscout-0.8.2.drv
         ```

fabaff committed:
jxscout: init at 0.8.2

Tool to perform JavaScript analysis for security researchers

https://github.com/francisconeves97/jxscout

fabaff commented on pkgs/by-name/jx/jxscout/package.nix:
```diff
@@ -0,0 +1,33 @@
+    "-s"
+    "-w"
+  ];
+
```
At first glance, `gitUpdater` would be the better choice.

fabaff commented:
It looks like that `npm`, `prettier` and `bun` should be provided by `jxscout` to skip their installation during runtime.

nicolas-goudry reviewed (commented):
@fabaff Feel free to ping me when you're done with this PR. I’ll review it then.
